.github/workflows/push.yml

```diff
@@ -42,8 +61,26 @@ jobs:
     steps:
       - name: Checkout the Repository
         uses: actions/checkout@v3
-        
-      - name: Install Dependencies
+
+      - name: Cache node modules
+        id: cache-npm
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: |
+            ~/.npm
+            node_modules
+          key: ${{ runner.os }}-generate-docs-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-generate-docs-${{ env.cache-name }}-
+            ${{ runner.os }}-generate-docs-
+            ${{ runner.os }}-
+
+      - if: ${{ steps.cache-npm.outputs.cache-hit != 'true' }}
+        name: List the state of node modules
+        continue-on-error: true
+        run: npm list
         run: yarn install --legacy-peer-deps
```

Comment:
Remove the conditional check, rename it to `Install dependencies`, and change the command to `npm install --legacy-peer-deps`.